subsys/bluetooth/mesh/rpl.c

```diff
@@ -118,34 +124,33 @@ bool bt_mesh_rpl_check(struct bt_mesh_net_rx *rx,
 	}
 
 	for (i = 0; i < ARRAY_SIZE(replay_list); i++) {
-		struct bt_mesh_rpl *rpl = &replay_list[i];
+		rpl = &replay_list[i];
 
 		/* Empty slot */
 		if (!rpl->src) {
-			if (match) {
-				*match = rpl;
-			} else {
-				bt_mesh_rpl_update(rpl, rx);
-			}
-
-			return false;
+			goto match;
 		}
 
 		/* Existing slot for given address */
 		if (rpl->src == rx->ctx.addr) {
+			if (!rpl->old_iv &&
+			    atomic_test_bit(&rpl_flags, PENDING_RESET) &&
+			    !atomic_test_bit(store, i)) {
+				/* Until rpl reset is finished, entry with old_iv == false and
```

Comment:
Could you clarify it a bit more? Why does it happen for `old_iv == false` and doesn't for `old_iv == true`?

Reply:
`store` is the map of `old_iv` flags. If `old_iv` is false then `store` is set. Does it have sense to check both `!rpl->old_iv` and `!atomic_test_bit(store, i)`?